repo/fsrepo/defaultds.go

```diff
@@ -13,22 +14,30 @@ import (
 	repo "github.com/ipfs/go-ipfs/repo"
 	config "github.com/ipfs/go-ipfs/repo/config"
 	"github.com/ipfs/go-ipfs/thirdparty/dir"
+
+	multi "github.com/ipfs/go-ipfs/Godeps/_workspace/src/github.com/ipfs/go-datastore/multi"
+	filestore "github.com/ipfs/go-ipfs/filestore"
 )
 
 const (
 	leveldbDirectory = "datastore"
 	flatfsDirectory  = "blocks"
+	fileStoreDir     = "filestore-db"
+	fileStoreDataDir = "filestore-data"
 )
+const useFileStore = true
+
+var _ = io.EOF
 
-func openDefaultDatastore(r *FSRepo) (repo.Datastore, error) {
+func openDefaultDatastore(r *FSRepo) (repo.Datastore, *filestore.Datastore, error) {
```

Comment:
I'm not sure on this, but i think that this might be cleaner if we construct the filestore outside of this function, and then pass it in as a parameter. Doing so would move the `useFileStore` 'global' check up a call or two, making it easier to abstract that out.